client.go

```diff
@@ -39,6 +51,15 @@ func (c *Client) DoClientRequest(ctx context.Context, client HTTPClient, url str
 	header.Set("Accept", "application/protobuf")
 	header.Set("Content-Type", "application/protobuf")
 
+	for i := range c.plugins {
+		p := c.plugins[i]
+		pluginHeaders := p.GetHeader(reqBodyBytes)
```

Comment:
Nit: Should that be 'pluginHeader' instead?

Reply:
I think that would be better, let me fix that